lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -21,23 +22,27 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
-  def self.upgrade_message
-    Puppet.warning <<-UPGRADE
-The current Puppet version is outdated and uses a library that was
-previously necessary on the current Ruby verison to support a colored console.
-
-Unfortunately this library prevents the DSC module from using a shared
-PowerShell process to dramatically improve the performance of resource
-application.
-
-To enable these improvements, it is suggested to upgrade to any x64 version of
-Puppet (including 3.x), or to a Puppet version newer than 3.x.
-    UPGRADE
-  end
-
+  POWERSHELL_UPGRADE_MSG = <<-UPGRADE
```

Comment:
Looks like you have superseded the proposed work in #248 / MODULES-4014 - which is great!

However, I'm not sure that this message makes the most sense for the DSC module.  DSC module requires PowerShell 5 since we're not using the `DscConfiguration` style cmdlets.

I think we can get away with something simpler here that is more specific to the DSC module (and I think as a result you can close out that other PR and the MODULES-4014 ticket at the same time)